bindings/matrix-sdk-ffi/src/encryption.rs

```diff
@@ -399,6 +399,10 @@ impl Encryption {
         listener: Box<dyn VerificationStateListener>,
     ) -> Arc<TaskHandle> {
         let mut subscriber = self.inner.verification_state();
+
+        // Reset the subscriber so the current observer value is emitted to the listener
+        subscriber.reset();
```

Comment:
No, that's not what I meant: I meant calling `subscribe_reset()` in place of `subscribe()` here: https://github.com/matrix-org/matrix-rust-sdk/blob/8865e2ff7463ff0b520f53f1a3ffc5500fbb3d86/crates/matrix-sdk/src/encryption/mod.rs#L777

But @poljar might have a different opinion when back, and I'm not clear about the implications of doing that. Let's wait for him to come back (i.e. revert to the previous state).

Reply:
Ok, reverted the fixup commit and left it as it was before. Let's wait for him.

Reply:
Yes, I think using `subscribe_reset()` would be the correct approach here, I think that re-implementing `subscribe_reset()` using `get()` and `subscribe()` might introduce a small race where we might miss an update.